### PR TITLE
🛡️ Sentinel: [MEDIUM] Fix sensitive data leak in CLI argument errors

### PR DESCRIPTION
**Pull request by rfv-code**
🚨 **Severity:** MEDIUM
💡 **Vulnerability:** Unsanitized CLI argument parsing errors leaked sensitive data to standard output and error. If a user provided a malformed JSON payload containing credentials (e.g., `--params '{"password": "secret"invalid'`), the underlying JSON/Zod error message included the raw fragment with the password.
🎯 **Impact:** Potential leakage of credentials into terminal logs, CI/CD systems, or upstream orchestrators that capture connector stderr/stdout.
🔧 **Fix:** Modified `writeArgErrorEnvelope` in `src/toolkit/connector.ts` to sanitize the error `message` using `scrubSecrets()` before constructing the error envelope and writing to standard streams.
✅ **Verification:** Verified that tests pass via `pnpm test tests/toolkit/connector.test.ts`. Manually verified that running the CLI script with an invalid password JSON string outputs `[REDACTED]` instead of the raw password string.

---
*PR created automatically by Jules for task [9956182532473670160](https://jules.google.com/task/9956182532473670160) started by @rfv-code*

### CODE REVIEW

File: src/toolkit/connector.ts [hunk 1]
```diff
@@ -561,15 +561,16 @@ export function createConnector<TSdk = unknown>(
     // hit the case where stdout is empty and the failure is text on stderr.
     // Exit code is 2 (CLI misuse), distinct from 1 (handled action-level error).
     const writeArgErrorEnvelope = (action: string, message: string): void => {
+      const scrubbed = scrubSecrets(message);
```
[the remark this answers is not in the document]
Comment (rfv-code) [reply src/toolkit/connector.ts:564]: Confirmed on this branch — and already fixed on the canonical.

I measured both sides rather than reading the regexes.

**On this branch (P1 is real).** This PR calls `scrubSecrets` at the `writeArgErrorEnvelope` site but does not touch `src/toolkit/audit/writer.ts`, so it inherits `main`'s scrubber, which has only four patterns (`SQUOTE`, `DQUOTE`, `AUTH_QUOTED`, `AUTH_LINE`) and no unquoted-value branch. Running `main`'s `scrubSecrets` on the exact case:

```
JSON.parse("password=secret") -> Unexpected token 'p', "password=secret" is not valid JSON
main scrubSecrets           -> Unexpected token 'p', "password=secret" is not valid JSON
```

The credential survives verbatim into both the stdout envelope and the stderr line. Your finding is correct as written.

**On #154's preservation branch (`fix/connector-credential-redaction`, tip `eb251df`), it is already closed.** That branch adds `SENSITIVE_UNQUOTED_RE`, whose separator class is `[:=]`, so it catches the bare `key=value` form:

```
main scrubSecrets           -> Unexpected token 'p', "password=secret" is not valid JSON
preservation scrubSecrets   -> Unexpected token 'p', "password="[REDACTED]"" is not valid JSON
```

Other unquoted shapes on that branch: `token=abc123` -> `token="[REDACTED]"`, `api_key=sk-live-9999` -> `api_key="[REDACTED]"`, `{"password":hunter2}` -> `{"password":"[REDACTED]"}`. The regression tests are in `tests/toolkit/audit.test.ts` ("redacts unquoted values", "redacts the unquoted fragment V8 echoes in JSON.parse messages").

Resolving here and closing this PR as a duplicate of #154. This finding is exactly why the narrow call-site-only version of this fix is not safe on its own.

**Comment by rfv-code** [comment]
Closing as duplicate of #154.

This PR's entire source change — `const scrubbed = scrubSecrets(message);` in `writeArgErrorEnvelope`, applied to both the stdout envelope and the stderr line — is **already present verbatim** on #154's preservation branch `fix/connector-credential-redaction` (`src/toolkit/connector.ts:663-674`, tip `eb251df`). This branch is a strict subset: #154 covers 8 exposure paths with 19 regression tests, this one covers 1 with none.

The Codex P1 on this PR is the concrete reason the narrow version is not safe standing alone, and I measured it both ways (see the thread, now resolved):

- **This branch** does not touch `src/toolkit/audit/writer.ts`, so it inherits `main`'s four-pattern scrubber with no unquoted-value branch. `password=secret` survives `scrubSecrets` unchanged and reaches stdout and stderr.
- **#154's branch** adds `SENSITIVE_UNQUOTED_RE` (separator class `[:=]`), which redacts it to `password="[REDACTED]"`, with tests in `tests/toolkit/audit.test.ts`.

So this PR applies the redaction call at the right place but without the scrubber upgrades that make the call effective — it leaves exactly the hole Codex found.

Note for maintainers: #154's own PR branch is not the branch to merge. See the standing note on that PR — the preservation branch `fix/connector-credential-redaction` holds the complete work.

If you believe this branch covers ground #154 does not, please cherry-pick onto `fix/connector-credential-redaction`.
